sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableTransactionAction.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables.models;
+
+/**
+ * Defines an transaction action to be included as part of a batch operation.
+ */
+public class TableTransactionAction {
+    private final TableTransactionActionType actionType;
+    private final TableEntity entity;
+    private final boolean ifUnchanged;
+
+    /**
+     * Initializes a new instance of the {@link TableTransactionAction}.
+     *
+     * @param actionType The operation type to be applied to the {@code entity}.
+     * @param entity The table entity to which the {@code actionType} will be applied.
+     */
+    public TableTransactionAction(TableTransactionActionType actionType, TableEntity entity) {
+        this(actionType, entity, false);
+    }
+
+    /**
+     * Initializes a new instance of the {@link TableTransactionAction}.
+     *
+     * @param actionType The operation type to be applied to the {@code entity}.
+     * @param entity The table entity to which the {@code actionType} will be applied.
+     * @param ifUnchanged When {@code true}, the ETag of the provided entity must
+     * match the ETag of the entity in the Table service. If the values do not match, the update will not occur and
+     * an exception will be thrown. This value is only applied for update operations.
```

Comment:
This should apply to Delete operations also.

Reply:
You mean having the `ifUnchanged` argument?

Reply:
correct - I believe that operation can take an etag parameter.

Reply:
Oh yes, we take that in our `TableClient` and `TableAsyncClient`. I will make sure the TableTransactionAction documentation reflects the actual behavior of when a user passes an `eTag` that does not match. By the way, I saw a bunch of other services name the ETag paramter `ifMatch`, should we consider doing the same?

Reply:
Also, when looking at .NET's code I couldn't help but wonder what would happen if a user creates a `TableTransactionAction` with an ETag that's different than the one in the `Entity` field? We can encounter the same issue in Java if I were to add such a parameter to our constructor.

Reply:
Talked to Chris separately and he clarified that the value specified in the input parameters is the one considered. The ETag in a `TableEntity` gets ignored.